pandas/core/indexes/datetimelike.py

```diff
@@ -606,9 +606,12 @@ def _get_delete_freq(self, loc: int):
                     loc = lib.maybe_indices_to_slice(
                         np.asarray(loc, dtype=np.intp), len(self)
                     )
-                if isinstance(loc, slice) and loc.step in (1, None):
-                    if loc.start in (0, None) or loc.stop in (len(self), None):
-                        freq = self.freq
+                if (
+                    isinstance(loc, slice)
+                    and loc.step in (1, None)
+                    and (loc.start in (0, None) or loc.stop in (len(self), None))
```

Comment:
this looks similar to is_full_slice

Reply:
true, but `is_full_slice` would return `False` if we had e.g.
```
>>> loc
slice(0, 3, 1)
>>> len(self)
10
```
while this condition would pass